### PR TITLE
Adding required fields for Log Source App

### LINKED ISSUE #19173: [spec] log source app cluster definitions not up to spec
#### Problem

```
./scripts/idl_lint.py --log-level warn ./examples/log-source-app/log-source-common/log-source-app.matter
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: EP0:GeneralCommissioning does not expose RegulatoryConfig(2) attribute at ./examples/log-source-ap
p/log-source-common/log-source-app.matter:416:3
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: EP0:GeneralCommissioning does not expose LocationCapability(3) attribute at ./examples/log-source-
app/log-source-common/log-source-app.matter:416:3
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: EP0:NetworkCommissioning does not expose MaxNetworks(0) attribute at ./examples/log-source-app/log
-source-common/log-source-app.matter:424:3
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: EP0:NetworkCommissioning does not expose Networks(1) attribute at ./examples/log-source-app/log-so
urce-common/log-source-app.matter:424:3
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: EP0:NetworkCommissioning does not expose InterfaceEnabled(4) attribute at ./examples/log-source-ap
p/log-source-common/log-source-app.matter:424:3
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: EP0:NetworkCommissioning does not expose LastNetworkingStatus(5) attribute at ./examples/log-sourc
e-app/log-source-common/log-source-app.matter:424:3
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: EP0:NetworkCommissioning does not expose LastNetworkID(6) attribute at ./examples/log-source-app/l
og-source-common/log-source-app.matter:424:3
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: EP0:NetworkCommissioning does not expose LastConnectErrorValue(7) attribute at ./examples/log-sour
ce-app/log-source-common/log-source-app.matter:424:3
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: EP0:OperationalCredentials does not expose NOCs(0) attribute at ./examples/log-source-app/log-sour
ce-common/log-source-app.matter:434:3
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: EP0:OperationalCredentials does not expose CurrentFabricIndex(5) attribute at ./examples/log-sourc
e-app/log-source-common/log-source-app.matter:434:3
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: Endpoint 0 does not expose cluster Basic (40)
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: Endpoint 0 does not expose cluster GroupKeyManagement (63)
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: Endpoint 0 does not expose cluster AdministratorCommissioning (60)
2022-06-03 10:14:26 ERROR   ERROR: Required attributes: Endpoint 0 does not expose cluster GeneralDiagnostics (51)
2022-06-03 10:14:26 ERROR   ERROR: Required commands: Cluster OperationalCredentials does not define mandatory command UpdateNOC(7) at ./examples/log-sour
ce-app/log-source-common/log-source-app.matter:310:1
2022-06-03 10:14:26 ERROR   Found 15 lint errors
```

### PR DESCRIPTION
**Pull request by AlexPygin**
Fixes linting errors called out by:

`./scripts/idl_lint.py --log-level warn ./examples/log-source-app/log-source-common/log-source-app.matter`

Errors:
```
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: EP0:GeneralCommissioning does not expose RegulatoryConfig(2) attribute at ./examples/log-source-app/log-source-common/log-source-app.matter:445:3
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: EP0:GeneralCommissioning does not expose LocationCapability(3) attribute at ./examples/log-source-app/log-source-common/log-source-app.matter:445:3
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: EP0:NetworkCommissioning does not expose MaxNetworks(0) attribute at ./examples/log-source-app/log-source-common/log-source-app.matter:453:3
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: EP0:NetworkCommissioning does not expose Networks(1) attribute at ./examples/log-source-app/log-source-common/log-source-app.matter:453:3
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: EP0:NetworkCommissioning does not expose InterfaceEnabled(4) attribute at ./examples/log-source-app/log-source-common/log-source-app.matter:453:3
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: EP0:NetworkCommissioning does not expose LastNetworkingStatus(5) attribute at ./examples/log-source-app/log-source-common/log-source-app.matter:453:3
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: EP0:NetworkCommissioning does not expose LastNetworkID(6) attribute at ./examples/log-source-app/log-source-common/log-source-app.matter:453:3
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: EP0:NetworkCommissioning does not expose LastConnectErrorValue(7) attribute at ./examples/log-source-app/log-source-common/log-source-app.matter:453:3
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: EP0:OperationalCredentials does not expose NOCs(0) attribute at ./examples/log-source-app/log-source-common/log-source-app.matter:463:3
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: EP0:OperationalCredentials does not expose CurrentFabricIndex(5) attribute at ./examples/log-source-app/log-source-common/log-source-app.matter:463:3
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: Endpoint 0 does not expose cluster BasicInformation (40)
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: Endpoint 0 does not expose cluster GroupKeyManagement (63)
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: Endpoint 0 does not expose cluster AdministratorCommissioning (60)
2023-02-10 19:39:37 ERROR   ERROR: Required attributes: Endpoint 0 does not expose cluster GeneralDiagnostics (51)
2023-02-10 19:39:37 ERROR   ERROR: Required commands: Cluster OperationalCredentials does not define mandatory command UpdateNOC(7) at ./examples/log-source-app/log-source-common/log-source-app.matter:330:1
2023-02-10 19:39:37 ERROR   Found 15 lint errors
```

Closes #19173

### CODE REVIEW

File: examples/log-source-app/log-source-common/log-source-app.matter [hunk 1]
```diff
@@ -190,6 +295,12 @@ server cluster NetworkCommissioning = 49 {
     int8u lqi = 7;
   }
 
+  readonly attribute access(read: administer) int8u maxNetworks = 0;
+  readonly attribute access(read: administer) NetworkInfo networks[] = 1;
+  attribute access(write: administer) boolean interfaceEnabled = 4;
+  readonly attribute access(read: administer) nullable NetworkCommissioningStatus lastNetworkingStatus = 5;
+  readonly attribute access(read: administer) nullable octet_string<32> lastNetworkID = 6;
+  readonly attribute access(read: administer) nullable int32s lastConnectErrorValue = 7;
```
Comment (bzbarsky-apple) [inline examples/log-source-app/log-source-common/log-source-app.matter:303]: Does this app actually implement these pieces?

File: examples/log-source-app/log-source-common/log-source-app.matter [hunk 2]
```diff
@@ -442,15 +783,48 @@ endpoint 0 {
     ram      attribute clusterRevision default = 1;
   }
 
+  server cluster BasicInformation {
+    callback attribute dataModelRevision default = 10;
+    callback attribute vendorName;
```
Comment (bzbarsky-apple) [inline examples/log-source-app/log-source-common/log-source-app.matter:788]: Are all these attributes that are being enabled actually implemented in these apps?

File: examples/log-source-app/log-source-common/log-source-app.matter [hunk 3]
```diff
@@ -442,15 +783,48 @@ endpoint 0 {
...
     ram      attribute clusterRevision default = 1;
   }
 
   server cluster NetworkCommissioning {
+    ram      attribute maxNetworks;
+    callback attribute networks;
+    ram      attribute interfaceEnabled;
+    ram      attribute lastNetworkingStatus;
+    ram      attribute lastNetworkID;
+    ram      attribute lastConnectErrorValue;
```
Comment (bzbarsky-apple) [inline examples/log-source-app/log-source-common/log-source-app.matter:827]: Likewise, is this actually implemented?   I certainly don't see any implementation of this in log-source-app.

File: examples/log-source-app/log-source-common/log-source-app.matter [hunk 4]
```diff
@@ -460,11 +834,47 @@ endpoint 0 {
     ram      attribute clusterRevision default = 1;
   }
 
+  server cluster GeneralDiagnostics {
+    callback attribute networkInterfaces;
+    callback attribute rebootCount;
+    callback attribute upTime;
```
Comment (bzbarsky-apple) [inline examples/log-source-app/log-source-common/log-source-app.matter:840]: Same question: are these bits actually implemented?

**Review by bzbarsky-apple — COMMENTED** [review]
We shouldn't be pretending like we implement pieces that are not actually implemented, in general...

**Comment by AlexPygin** [comment]
I'm going to close this PR because I don't have an easy way to validate these changes (Don't have good development environment).

**Pull request closed without merging**